### DISCUSSION TITLE
Storing parameters in case of shutdown

### DISCUSSION THREAD

**Discussion by guidoSNT** [yamcs/yamcs#925] [Q&A] [answered]
Hi Yamcs team,

I've been trying to save data from yamcs in case of failure or shutdown of the system but I have an issue with the logs it makes. I can save the files stored in `target/yamcs/yamcs-data/<instance>.rdb` but I'm not familiar with RocksDB so I don't know if I can rebuild the timeline of parameters from this.

So the questions are:

- Is this the best method to save data in case of failure or should I do something different?
- Can I reconstruct the parameters with the RocksDB logs?

Thanks!

**Comment by xpromache** [comment]
You need Yamcs to read the data from the RocksDB files.  If you have the RocksDB files, then you can run Yamcs somewhere else to retrieve the data, can't you?

Otherwise you can export in CSV or something like that but it will be difficult to have a complete copy.

**Comment by guidoSNT** [comment]
I would like to retrieve the data from the RocksDB files to Yamcs but I don't know if it's possible. Can I pass Yamcs a complete copy of the RocksDB without having to somehow translate this database to packets again? If it is, how can it be done?

**Reply by xpromache** [reply]
I don't understand what you want to do.

Where is the RocksDB database coming from, not from Yamcs?

**Reply by guidoSNT** [reply, marked as answer]
Hi, sorry for taking so long. In the end we decided that the best option is to rely on the Parameters -> <parameter> -> Historical Data -> export CSV.

Thanks
